xchange-examples/src/main/java/org/knowm/xchange/examples/blockchain/BlockchainAddressDemo.java

```diff
@@ -1,32 +1,134 @@
+
   public static void main(String[] args) throws IOException {
+    System.out.println("===== MARKET DATA SERVICE =====");
+    marketDataServiceDemo();
+    System.out.println("===== ACCOUNT SERVICE =====");
+    accountServiceDemo();
+    System.out.println("===== TRADE SERVICE =====");
+    tradeServiceDemo();
+  }
 
-    Exchange blockchainExchangexchange =
-        ExchangeFactory.INSTANCE.createExchange(BlockchainExchange.class);
-    Blockchain blockchain =
-        ExchangeRestProxyBuilder.forInterface(
-                Blockchain.class, blockchainExchangexchange.getExchangeSpecification())
-            .build();
+  private static void marketDataServiceDemo() throws IOException {
+    MarketDataService marketDataService = BLOCKCHAIN_EXCHANGE.getMarketDataService();
+
+    System.out.println("===== getTicker =====");
+    Ticker ticker = marketDataService.getTicker(CurrencyPair.ETH_BTC);
+    System.out.println(objectMapper.writeValueAsString(ticker));
+
+    System.out.println("===== getTickers =====");
+    List<Ticker> tickers = marketDataService.getTickers(null);
+    System.out.println(objectMapper.writeValueAsString(tickers));
+
+    System.out.println("===== getOrderBook =====");
+    OrderBook orderBook = marketDataService.getOrderBook(CurrencyPair.ETH_BTC);
+    System.out.println(objectMapper.writeValueAsString(orderBook));
+
+    System.out.println("===== getTrades =====");
+    Trades trades = marketDataService.getTrades(CurrencyPair.ETH_BTC);
+    System.out.println(objectMapper.writeValueAsString(trades));
+  }
+
+  private static void accountServiceDemo() throws IOException {
+    AccountService accountService = BLOCKCHAIN_EXCHANGE.getAccountService();
+
+    System.out.println("===== withdrawFunds =====");
+    String withdraw = accountService.withdrawFunds(Currency.BTC, BigDecimal.valueOf(0.00005), "ea1f34b3-e77a-4646-9cfa-5d6d3518c6d3");
+    System.out.println(objectMapper.writeValueAsString(withdraw));
+
+    System.out.println("===== requestDepositAddress =====");
+    String address = accountService.requestDepositAddress(Currency.ETH);
+    System.out.println(objectMapper.writeValueAsString(address));
+
+    System.out.println("===== requestDepositAddressData =====");
+    AddressWithTag addressWithTag = accountService.requestDepositAddressData(Currency.ETH);
+    System.out.println(objectMapper.writeValueAsString(addressWithTag));
+
+    System.out.println("===== getFundingHistory =====");
+    TradeHistoryParams params = accountService.createFundingHistoryParams();
 
-    BitcoinAddress bitcoinAddress = blockchain.getBitcoinAddress("XXX");
-    System.out.println(bitcoinAddress.toString());
+    if (params instanceof TradeHistoryParamsTimeSpan) {
+      final TradeHistoryParamsTimeSpan timeSpanParam = (TradeHistoryParamsTimeSpan) params;
+      timeSpanParam.setStartTime(
+              new Date(System.currentTimeMillis() - (1 * 12 * 30 * 24 * 60 * 60 * 1000L)));
```

Comment:
what does this calculation mean? Move it to a constant

Reply:
is to set the time to obtain the funds from a start and end time.